src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -672,10 +672,16 @@ protected function dumpContainer(ConfigCache $cache, ContainerBuilder $container
      *
      * @param ContainerInterface $container The service container
      *
+     * @throws \InvalidArgumentException When the container is not an instance of ContainerBuilder
+     *
      * @return DelegatingLoader The loader
      */
     protected function getContainerLoader(ContainerInterface $container)
     {
+        if (!$container instanceof ContainerBuilder) {
+            throw new \InvalidArgumentException(sprintf('The container must be an instance of Symfony\Component\DependencyInjection\ContainerBuilder, got %s.', get_class($container)));
```

Comment:
Fair enough, i tend to look at it more abstract. Ie. the current design is semi-flawed, and could be fixed imo. I dont like to depend on PHP what will happen..

>  why not just change the phpdoc

Not sure.. imo. it's more confusing.

Lets keep it as is then :confused: